src/main/java/teammates/ui/constants/ActionNames.java

```diff
@@ -0,0 +1,115 @@
+package teammates.ui.constants;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * Action names for http requests.
+ */
+public enum ActionNames {
```

Comment:
Even though this works, this adds an unnecessary coupling to the list of actions already defined somewhere else. If we decide to add new action, rename some actions, deprecate/remove some actions, this enum's accuracy is compromised if not kept in sync. 

Better alternative: create an admin-only (later on, admin+maintainer-only) API endpoint that returns the list of actions. It is straightforward to get this list populated from the other place that already listed this down. (hint: you're looking at the right place if you find out that you need to (1) change a field from `private` to package level and (2) flatten a nested map)

Reply:
I have removed this file and action names are now retrieved by flattening the map in ActionFactory. The backend code will be added to #11269